gen3users/conversion.py

```diff
@@ -0,0 +1,425 @@
+        user_yaml_dict {dict} -- Contents of a user.yaml file.
+    """
+    for permissions, name in BASIC_ROLES.items():
+        user_yaml_dict["rbac"]["roles"].append(
+            {
+                "id": name,
+                "permissions": [
+                    {
+                        "id": PRIVILEGE_TO_ROLE_NAME[p],
+                        "action": {"service": "*", "method": p},
+                    }
+                    for p in permissions
+                ],
+            }
+        )
+
+
+def auth_id_to_resource_path(user_yaml_dict, auth_id):
+    """
+    Recursively searches the resource tree for a program or project
+    named auth_id.
+
+    Arguments:
+        user_yaml_dict {dict} -- Contents of a user.yaml file.
+        auth_id {str} -- Program or project name as defined in user access
+
+    Returns:
+        [str] -- resource path for this auth_id if it exist, None otherwise
+    """
+    for resource in user_yaml_dict["rbac"].get("resources", []):
+        res = auth_id_to_resource_path_recursive("", resource, auth_id)
+        if res:
+            return res
+
+
+def auth_id_to_resource_path_recursive(root, resource, auth_id):
+    """
+    Recursively searches the resource tree for a program or project
+    named auth_id.
+
+    Arguments:
+        root {str} -- current resource path
+        resource {dict} -- current resource.
+        auth_id {str} -- Program or project name as defined in user access
+
+    Returns:
+        [str] -- resource path for this auth_id if it exist, None otherwise
+    """
+    new_root = "{}/{}".format(root, resource["name"])
+    if resource["name"] in ["programs", "projects"]:
+        for sub in resource.get("subresources", []):
+            if sub["name"] == auth_id:
+                return "{}/{}".format(new_root, sub["name"])
+    for sub in resource.get("subresources", []):
+        res = auth_id_to_resource_path_recursive(new_root, sub, auth_id)
+        if res:
+            return res
+
+
+def convert_old_user_yaml_to_new_user_yaml(user_yaml, dest_path=None):
+    """
+    Converts a user.yaml file to the new format required by the
+    latest fence and arborist.
+
+    Arguments:
+        user_yaml {str} -- Contents of a user.yaml file.
+
+    Keyword Arguments:
+        dest_path {str} -- Optional destination file. (default: {None})
+
+    Returns:
+        [str] -- Contents of the new user.yaml file.
+    """
+    # make sure the syntax is valid and expected fields exist
+    print("Pre-conversion validation:")
+    validate_user_yaml(user_yaml)
+
+    print("Converting...")
+    old_user_yaml = yaml.safe_load(user_yaml)
+    new_user_yaml = {
+        "cloud_providers": old_user_yaml.get("cloud_providers", {}),
+        "groups": old_user_yaml.get("groups", {}),
+        "users": {},
+        "rbac": {
+            "groups": [],
+            "anonymous_policies": ["open_reader"],
+            "all_users_policies": ["open_reader"],
+            "policies": [
```

Comment:
just for the sake of organization, I would consider putting this at the top level as a global so it's clear this is just static/global config

Reply:
(even in its own json file would work)